src/integrations/watchers/indexable-post-watcher.php

```diff
@@ -172,8 +183,6 @@ public function updated_indexable( $indexable, $post ) {
 		}
 
 		$this->update_relations( $post );
-
-		$indexable->save();
```

Comment:
This one worries me a bit, since I cannot really oversee the impact. Do we know for sure that the save here was not needed to make sure the relations were also saved in the original indexable or something?

Reply:
> Do we know for sure that the save here was not needed to make sure the relations were also saved in the original indexable or something?

If I'm reading [the PR that introduced this](https://github.com/Yoast/wordpress-seo/pull/16179/files#diff-05bc2ee71341baec8defb1436db3ee42b9efed75530526f7127e5f6d0dae85dcR175) right, this was added to save the updated link count, which we already coming into this function, in the `build_indexable()` function.

Basically, like I'm saying in the technical choices:
> * [A redundant db write query was removed](https://github.com/Yoast/wordpress-seo/pull/21335/commits/f7759b74a547524509aece00cbc10529e5d9748f) when a post is saved, because the indexable has already been saved coming into that function and the only thing the second write did was to update the updated_at column of the indexable (that had already been updated before coming into the function)

But, since it can be a pretty sneaky change I re-did my impact check and did the following
* Confirmed that the `updated_indexable()` is not used anywhere in our codebases, only in the `build_indexable()` one, which already did a `save_indexable()` before calling that one
* I also re-instated that line and compared the queries in one post save run, which were respectively:
```
UPDATE `wp_yoast_indexable` SET `has_ancestors` = '0', `link_count` = '0', `permalink` = 'http://main.local/engtest/', `permalink_hash` = '26:4196e57c1ffd32e796f4cefc6c269e8d', `updated_at` = '2024-05-24 07:00:29' WHERE `id` = '1'
```
and 
```
UPDATE `wp_yoast_indexable` SET `permalink` = 'http://main.local/engtest/', `permalink_hash` = '26:4196e57c1ffd32e796f4cefc6c269e8d', `updated_at` = '2024-05-24 07:12:51' WHERE `id` = '1'
```
so we can see that the second save does nothing of significance.
* I even changed the link count and re-saved (with that removed `$indexable->save();` line temporarily re-added) and the query that it resulted in was again:
```
UPDATE `wp_yoast_indexable` SET `permalink` = 'http://main.local/engtest/', `permalink_hash` = '26:4196e57c1ffd32e796f4cefc6c269e8d', `updated_at` = '2024-05-24 07:12:51' WHERE `id` = '1'
```
so again, of no significance.


Now, I'm not married to the removal of the extra query if you still have concerns about it, but let me know if the above put you more at ease 😅

Reply:
No concerns anymore makes sense :)